### PR TITLE
feat(amplify-util-uibuilder): amplify js backwards compatibility

### PR DESCRIPTION
**Pull request by letsbelopez**
#### Description of changes

Update the StartCodegenJob operation to be backwards compatible for amplify js. StartCodegenJob is now passed the relevant local project dependencies for codegen to know which version of amplify js to codegen.

E.g. If the local project has a dependecy of `aws-amplify ^5.0.2`, then codegen will generate v5 compatible code. If there are no dependencies yet, codegen will generate the latest code available and recommend to install the dependencies.

#### Issue #, if available

NA

#### Description of how you validated changes

Unit test
Testing against a personal app
Currently running e2e tests, I'll update when they complete here.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#tests)
- [x] Relevant documentation is changed or added (and PR referenced)
- [x] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies
- [x] [Pull request labels](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#labels) are added

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-cli-npm/index.ts [hunk 1]
```diff
@@ -16,4 +16,4 @@ export const install = async (): Promise<void> => {
   return binary.install();
 };
 
-// force version bump to 12.5.0
```
Comment (sobolk) [inline packages/amplify-cli-npm/index.ts:19]: if you merge this as is version will be `12.6` given `feat` in the PR description

File: packages/amplify-util-uibuilder/src/commands/utils/notifyMissingPackages.ts [hunk 2]
```diff
@@ -3,44 +3,82 @@ import { printer } from '@aws-amplify/amplify-prompts';
...
+import { CodegenDependencies } from 'aws-sdk/clients/amplifyuibuilder';
 
-const getRequiredDependencies = (hasStorageManagerField?: boolean): RequiredDependency[] =>
-  new ReactRequiredDependencyProvider().getRequiredDependencies(hasStorageManagerField);
+type PackageJson = { dependencies: { [key: string]: string } };
 
 /**
- * Displays a warning to the user if they have npm dependencies
- * they need to install in their application for UIBuilder components to work properly
+ * Returns a parsed package.json file
```
Comment (sobolk) [inline packages/amplify-util-uibuilder/src/commands/utils/notifyMissingPackages.ts:12]:
```suggestion
 * Returns a parsed package.json file if present
```

File: packages/amplify-util-uibuilder/src/commands/utils/notifyMissingPackages.ts [hunk 3]
```diff
@@ -3,44 +3,82 @@ import { printer } from '@aws-amplify/amplify-prompts';
...
+ * @param context cli context object
+ * @returns A JSON object representing the package.json file in the local project
```
Comment (sobolk) [inline packages/amplify-util-uibuilder/src/commands/utils/notifyMissingPackages.ts:14]:
```suggestion
 * @returns A JSON object representing the package.json file in the local project or undefined if not found
```

**Review by sobolk — COMMENTED** [review]
LGTM. Please provide passing e2e run.

**Review by sobolk — APPROVED** [review]

**Review by 0618 — DISMISSED** [review]

**Review by 0618 — APPROVED** [review]

**Review by sobolk — APPROVED** [review]

**Comment by Amplifiyer** [comment]
Is this a copy/duplicate of https://github.com/aws-amplify/amplify-cli/pull/13288

**Pull request closed without merging**